prio/encrypt.c

```diff
@@ -458,6 +458,9 @@ set_gcm_params(SECItem* paramItem, CK_GCM_PARAMS* param, unsigned char* nonce,
 
   param->pIv = nonce;
   param->ulIvLen = GCM_IV_LEN_BYTES;
+#ifdef CRYPTOKI_VERSION_MAJOR
```

Comment:
Thanks for taking the time to track this down!

I'm wondering if a safer option would be to have the SConstruct file define `NSS_PCKS11_2_0_COMPAT`. I'm worried that if we just check whether `CRYPTOKI_VERSION_MAJOR` is defined, we could potentially break the build on older versions of NSS (which also define `CRYPTOKI_VERSION_MAJOR`).

Reply:
`CRYPTOKI_VERSION_MAJOR` (value is 3) was introduced together with `NSS_PKCS11_2_0_COMPAT`. Older versions of NSS define `NSS_CKCAPI_CRYPTOKI_VERSION_MAJOR` (value is 2) instead. Using `NSS_PCKS11_2_0_COMPAT` forces non-scons consumers (e.g., Firefox) to define it as well.

I only want to drop  `allow_failures`. Whether and how to support old NSS is up to you.

Reply:
Actually, NSS >= 3.52 still defines `NSS_CKCAPI_CRYPTOKI_VERSION_MAJOR` (value is 2).